packages/network/src/solana/transactions.ts

```diff
@@ -52,3 +79,131 @@ export const simulateTransaction = async (
+	connection: Connection,
+	sender: PublicKey,
+	receiver: PublicKey,
+	amount: number,
+) => {
+	const instructions = [
+		SystemProgram.transfer({
+			fromPubkey: sender,
+			toPubkey: receiver,
+			lamports: amount,
+		}),
+	];
+
+	const blockhash = (await connection.getLatestBlockhash('finalized'))
+		.blockhash;
+
+	const message = new TransactionMessage({
+		payerKey: new PublicKey(sender),
+		recentBlockhash: blockhash,
+		instructions,
+	}).compileToV0Message();
+
+	return new VersionedTransaction(message);
+};
+
+export const constructSendSPLTokenTransactionInSol = async (
+	connection: Connection,
+	sender: PublicKey,
+	receiver: PublicKey,
+	amount: number,
+	token: Token | Collectible,
+) => {
+	// ATA
+	const mintAddress = new PublicKey(token.account.mint as string);
+	const senderATAddress = getAssociatedTokenAddressSync(mintAddress, sender);
+	const receiverATAddress = getAssociatedTokenAddressSync(
+		mintAddress,
+		receiver,
+	);
+	const receiverATA = await connection.getAccountInfo(receiverATAddress);
+
+	// PDA
+	const [metadata] = getMetadataPda(mintAddress);
+
+	const [edition] = getMasterEditionPda(mintAddress);
+
+	const [ownerTokenRecord] = getTokenRecordPda({
+		mint: mintAddress,
+		token: senderATAddress,
+	});
+	const [destinationTokenRecord] = getTokenRecordPda({
+		mint: mintAddress,
+		token: receiverATAddress,
+	});
+
+	const instructions = [];
+	if (!receiverATA) {
+		const createATAInstruction = createAssociatedTokenAccountInstruction(
+			sender,
+			receiverATAddress,
+			receiver,
+			mintAddress,
+		);
+		instructions.push(createATAInstruction);
+	}
+
+	if ('collectionId' in token) {
+		const metadataInfo = await connection.getAccountInfo(metadata);
+		let isProgrammableNft = false;
+		if (metadataInfo?.data) {
+			const [data] = Metadata.deserialize(metadataInfo.data);
+			const { tokenStandard } = data;
+			isProgrammableNft = isProgrammable({ tokenStandard });
+		}
+
+		const transferInstruction = createTransferNftInstruction(
+			{
+				token: senderATAddress,
+				tokenOwner: sender,
+				destination: receiverATAddress,
+				destinationOwner: receiver,
+				mint: mintAddress,
+				metadata,
+				payer: sender,
+				edition,
+				ownerTokenRecord: isProgrammableNft ? ownerTokenRecord : undefined,
+				destinationTokenRecord: isProgrammableNft
+					? destinationTokenRecord
+					: undefined,
+				splAtaProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+				splTokenProgram: TOKEN_PROGRAM_ID,
+				authority: sender,
+				sysvarInstructions: SYSVAR_INSTRUCTIONS_PUBKEY,
+				authorizationRulesProgram: TOKEN_AUTH_RULES_KEY,
+			},
+			{
+				transferArgs: {
+					__kind: 'V1',
+					amount,
+					authorizationData: null,
+				},
+			},
+		);
+		instructions.push(transferInstruction);
+	} else {
+		const transferInstruction = createTransferTokenInstruction(
+			senderATAddress,
+			receiverATAddress,
+			sender,
+			amount,
+		);
+		instructions.push(transferInstruction);
+	}
```

Comment:
I think we should keep the previous function and implement another one for the new standard, the checking standard needs to be in the upper scope which calls these functions.